tests/ctst/HOW_TO_WRITE_TESTS.md

```diff
@@ -0,0 +1,108 @@
+# How to write a test that runs in parallel
+
+Writting intergation / behavior tests that run in parallel is not trivial.
+This page lists all the rules to follow when writing a test that runs in
+parallel with other tests.
+
+*This is specific to Zenko, but the rules are applicable for any suite of*
+*tests in parallel.*
+
+## 1. Tests must be idempotent.
+
+In parallel, the tests can run in any order. It's not because a test works at a
+given time, that it will work if the test order changes after adding new ones
+or renaming files. You must ensure the scenarios are fully idempotent.
+"Flakies" are mostly due to non-idempotent tests, then bugs. It is rarely due
```

Comment:
> it is rarely due to the platform

Even though it's *actually* true, I'm not sure we want to write that in the documentation, as the test are made to verify that the platform works..

Reply:
By platform I mean the CI environment, I will make it clearer that we cannot always think it's due to limited resources in the CI, or some sort of slowness we cannot explain